boilerplate/openshift/custom-catalog-osd-operator/current-version-getter.sh

```diff
@@ -0,0 +1,167 @@
+
+REPO_ROOT=$(git rev-parse --show-toplevel)
+OPERATOR_NAME=$(basename ${REPO_ROOT}  | sed 's/-catalog//')
+CERT_DIR="${REPO_ROOT}/service-account/rhcs-request/"
+CERT_FILE_NAME="custom-catalog-source-index-builder"
+PYXIS_ENDPOINT="https://pyxis.engineering.redhat.com"
+VERSIONS_DIR="${REPO_ROOT}/versions"
+
+
+# compare_versions accepts two semvers.
+# Its return value indicates how they compare:
+# 0: the semvers are equal
+# 1: the first semver is newer
+# 2: the second semver is newer
+compare_versions() {
+    [[ $1 == $2 ]] && return 0
+    local older=$(printf "$1\n$2\n" | sort -V | head -1)
+    [[ $older == $2 ]] && return 1
+    return 2
+}
+
+function update_latest_version() {
+  # perform the API request
+  local channel_name=${1}
+  local version_in_repo=${2}
+  local version_file=${3}
+
+  local available_operator_versions=$(curl -X GET --key ${CERT_DIR}/${CERT_FILE_NAME}.key --cert ${CERT_DIR}/${CERT_FILE_NAME}.crt \
+    "${PYXIS_ENDPOINT}/v1/operators/bundles?page_size=100&organization=redhat-operators&filter=package==${OPERATOR_NAME};in_index_img==true;channel_name==${channel_name}" | \
+    jq -r "[.data[] | {bundle_image: .bundle_path, operator_version: .version, channel_name: .channel_name, operator_name: .package }]" )
+
+    jq -c '.[]' <<< ${available_operator_versions} | while read i; do
+      local upstream_operator_version=$( jq .operator_version <<< "${i}" )
+      upstream_operator_version=$(sed -e 's/^"//' -e 's/"$//' <<< ${upstream_operator_version})
+      
+      compare_versions ${upstream_operator_version} ${version_in_repo}
+      if [[ $? == 1 ]]; then
+        echo "Found newer version upstream: ${upstream_operator_version}"
+        echo ${i} |jq > ${version_file}
+      fi
+    done
+}
+
+function create_cert_dir_files() {
+  mkdir -p ${CERT_DIR}
+  if [[ -z "${PYXIS_CERT}" || -z "${PYXIS_KEY}" ]]; then
+    echo "Red hat certificates required for Pyxis API not found. Please set PYXIS_CERT & PYXIS_KEY with appropriate values and try again"
+    exit 1
+  fi
+  #Write cert files to disk for API calls
+  echo "${PYXIS_CERT}" > ${CERT_DIR}/${CERT_FILE_NAME}.crt
+  echo "${PYXIS_KEY}" > ${CERT_DIR}/${CERT_FILE_NAME}.key
+}
+
+function channel_exists_in_repo() {
+  local upstream_channel=${1}
+  for  f in ${VERSIONS_DIR}/* 
+  do
+    channel_name=$( cat ${f} | jq -r '.channel_name' )
+    if [[ ${channel_name} == ${upstream_channel} ]]; then
+      # found
+      return 0
+    fi
+  done
+  # Not Found
+  return 1
+}
+
+function update_version_files() {
+  # For each major channel version, figure out the latest operator version
+  for  f in ${VERSIONS_DIR}/* 
+  do
+    channel_name=$( cat ${f} | jq -r '.channel_name' )
+    version=$(cat ${f} | jq -r '.operator_version')
+    update_latest_version ${channel_name} ${version} ${f}
+  done
+}
+
+function commit_version_file_changes() {
+  local latest_operator_version=''
+  local version_file=''
+  if [[ -n $(git diff --name-only -q ${VERSIONS_DIR}) ]]; then
```

Comment:
What does `-q` do?

Reply:
`-q` was for quiet but the `--name-only` already only spits out the name as required so I will remove `-q`.